components/script/dom/readablestreamdefaultreader.rs

```diff
@@ -41,28 +45,81 @@ impl ReadableStreamDefaultReader {
         proto: Option<SafeHandleObject>,
         stream: DomRoot<ReadableStream>,
     ) -> Fallible<DomRoot<Self>> {
-        todo!()
+        Self::new_with_proto(global, proto, stream)
     }
 
-    fn new_inherited() -> ReadableStreamDefaultReader {
-        ReadableStreamDefaultReader {
-            reflector_: Reflector::new(),
-            read_requests: DomRefCell::new(Vec::new()),
-        }
+    pub fn new(global: &GlobalScope, stream: DomRoot<ReadableStream>) -> Fallible<DomRoot<Self>> {
```

Comment:
Do we still need this?

Reply:
I'm not sure about the pattern here. But most other DOM types have `new` fn  and call `new_with_proto` with `proto: None`

Reply:
Then `new_with_proto` below should be called `new` instead.